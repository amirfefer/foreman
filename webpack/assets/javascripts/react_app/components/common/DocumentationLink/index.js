import React from 'react';
import { MenuItem } from 'patternfly-react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

const DocumentationLink = ({ href, id }) => {
  const handleClick = e => {
    e.preventDefault();
    window.open(href, '_blank');
  };

  return (
    <MenuItem key="documentationUrl" id={id} href={href} onClick={handleClick}>
      <Icon type="question-sign" className="icon-black" />
      {` ${__('Documentation')}`}
    </MenuItem>
  );
};

export default DocumentationLink;

DocumentationLink.propTypes = {
  href: PropTypes.string.isRequired,
  id: PropTypes.string,
};

DocumentationLink.defaultProps = {
  id: 'documentationLink',
};
