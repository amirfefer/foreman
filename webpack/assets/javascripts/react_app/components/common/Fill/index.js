import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './FillActions';
import reducer from './FillReducer';

import Fill from './Fill';

// map state to props
const mapStateToProps = state => ({
  components: state.extendable,
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = { extendable: reducer };

// export connected component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Fill);
