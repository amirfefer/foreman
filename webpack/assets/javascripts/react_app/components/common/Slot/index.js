import { connect } from 'react-redux';
import { selectRegisteredFills } from './SlotSelectors';
import Slot from './Slot';

// map state to props
const mapStateToProps = (state, ownProps) => ({
  fills: selectRegisteredFills(state, ownProps.id),
});

// export connected component
export default connect(mapStateToProps)(Slot);
