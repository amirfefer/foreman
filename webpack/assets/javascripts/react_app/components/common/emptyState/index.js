import React from 'react';
import { EmptyState as PfEmptyState, Button } from 'patternfly-react';

const EmptyState = (props) => {
  const {
    icon = 'add-circle-o',
    header,
    description,
    documentation: {
      label = __('For more information please see'),
      buttonLabel = __('Documentation'),
      url,
    },
    action,
    secondayActions,
  } = props;

  return (
    <PfEmptyState>
      <PfEmptyState.Icon type="pf" name={icon} />
      <PfEmptyState.Title>{header}</PfEmptyState.Title>
      <PfEmptyState.Info>{description}</PfEmptyState.Info>
      {url &&
        <PfEmptyState.Help>
         {label}
          <a href={url}> {buttonLabel} </a>
        </PfEmptyState.Help>
      }
      <PfEmptyState.Action>
        <Button href={action.url} bsStyle="primary" bsSize="large">
          {action.title}
        </Button>
      </PfEmptyState.Action>
      {secondayActions && (
        <PfEmptyState.Action secondary>
          {secondayActions.map(item => (
            <Button href={action.url} title={action.title}>
              {item.title}
            </Button>
          ))}
        </PfEmptyState.Action>
      )}
    </PfEmptyState>
  );
};

export default EmptyState;

