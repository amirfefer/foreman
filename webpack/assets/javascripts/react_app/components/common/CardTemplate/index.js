import React from 'react';
import CardItem from './CardItem';

import { addGlobalFill } from '../../common/Fill/GlobalFill';

const CardTemplate = ({ content, header, slotID, cardID }) => {
  addGlobalFill(
    slotID,
    cardID,
    <CardItem content={content} header={header} />,
    200
  );
};

export default CardTemplate;
