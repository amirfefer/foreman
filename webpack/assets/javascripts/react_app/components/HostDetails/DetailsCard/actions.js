import { capitalize } from '../../../common/helpers';
import { translate as __, sprintf } from '../../../common/I18n';
import { put } from '../../../redux/API';
import { POWER_REQURST_KEY } from './constant';

export const changeHostPower = (state, hostID) =>
  put({
    key: POWER_REQURST_KEY,
    params: { power_action: state },
    url: `/api/hosts/${hostID}/power`,
    errorToast: err => __(`an error occured - ${err}`),
    successToast: () =>
      sprintf('Power has been set to "%s" successfully', state),
    updateData: (prevState, { power }) => {
      if (power)
        return {
          ...prevState,
          state,
          title: capitalize(state),
        };
      return { ...prevState };
    },
  });
