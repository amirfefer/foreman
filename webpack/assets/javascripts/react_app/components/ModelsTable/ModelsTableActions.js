import {
  HW_MODEL_TABLE_REQUEST,
  HW_MODEL_TABLE_SUCCESS,
  HW_MODEL_TABLE_FAILURE,
} from './ModelsTableConsts';
import { ajaxRequestAction } from '../../redux/actions/common';

export const getTableItems = (url, params, controller) => dispatch =>
  ajaxRequestAction({
    dispatch,
    requestAction: HW_MODEL_TABLE_REQUEST,
    successAction: HW_MODEL_TABLE_SUCCESS,
    failedAction: HW_MODEL_TABLE_FAILURE,
    url,
    params,
    item: { controller, url },
  });
