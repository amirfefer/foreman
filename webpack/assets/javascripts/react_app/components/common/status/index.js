import React from 'react';
import { connect } from 'react-redux';
import { get } from 'lodash';
import { Icon } from 'patternfly-react';
import { simpleLoader } from '../../common/Loader';
import * as StatusAction from '../../../redux/actions/status/';

export class Status extends React.Component {
  componentDidMount() {
    const { data: { id, type, url }, getStatus } = this.props;
    if (url) {
      getStatus({ id, type }, url);
    }
  }

  render() {
    const {
      status: {
        status, message, error, success,
      },
      getMessage,
    } = this.props;

    if (message && getMessage) {
      return <span>{message[getMessage]}</span>;
    }

    if (status || success !== undefined) {
      if (message && message.warning) {
        return (
          <div title={message.warning.message}>
            <Icon
              type='pf'
              name='warning-triangle-o'
            />
          </div>
        );
      }
      return (
        <div title={typeof message === 'string' ? message : undefined}>
          <Icon
            type='pf'
            name={status === 'OK' || success ? 'ok' : 'error-circle-o'}
          />
        </div>
      );
    }

    if (error) {
      return (
        <div title={message}>
          <Icon
            type='pf'
            name='error-circle-o' />
        </div>
      );
    }

    return simpleLoader('xs');
  }
}

const mapStateToProps = (state, ownProps) => {
  const { type, id } = ownProps.data;
  return {
    status: get(state.status, `${type}.${id}`) || {},
  };
};

export default connect(mapStateToProps, StatusAction)(Status);
