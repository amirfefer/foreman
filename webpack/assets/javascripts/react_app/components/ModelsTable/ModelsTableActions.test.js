import {
  HW_MODEL_TABLE_REQUEST,
  HW_MODEL_TABLE_SUCCESS,
  HW_MODEL_TABLE_FAILURE,
} from './ModelsTableConsts';
import { getTableItems } from './ModelsTableActions';
import { ajaxRequestAction } from '../../redux/actions/common';

jest.mock('../../redux/actions/common');
describe('ModelsTableActions', () => {
  it('sortTable should call ajaxRequestAction with url ', () => {
    const url = 'url';
    const dispatch = jest.fn();
    const expectedParams = {
      dispatch,
      failedAction: HW_MODEL_TABLE_FAILURE,
      requestAction: HW_MODEL_TABLE_REQUEST,
      successAction: HW_MODEL_TABLE_SUCCESS,
      url,
      params: {},
      item: { controller: 'models', url },
    };
    const dispatcher = getTableItems(url, {}, 'models');

    dispatcher(dispatch);
    expect(ajaxRequestAction).toBeCalledWith(expectedParams);
  });
});
