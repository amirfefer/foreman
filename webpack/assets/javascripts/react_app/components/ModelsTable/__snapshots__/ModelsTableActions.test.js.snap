// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 1`] = `
Object {
  "payload": Object {
    "results": Array [],
  },
  "type": "TABLE_INIT",
}
`;

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 2`] = `
Object {
  "payload": Object {
    "page": "3",
    "results": Array [],
  },
  "type": "TABLE_INIT",
}
`;

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 3`] = `
Object {
  "payload": Object {
    "perPage": "5",
    "results": Array [],
  },
  "type": "TABLE_INIT",
}
`;

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 4`] = `
Object {
  "payload": Object {
    "results": Array [],
    "searchQuery": "name = \\"model name 1\\"",
  },
  "type": "TABLE_INIT",
}
`;

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 5`] = `
Object {
  "payload": Object {
    "results": Array [],
  },
  "type": "TABLE_INIT",
}
`;

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 6`] = `
Object {
  "payload": Object {
    "results": Array [],
  },
  "type": "TABLE_INIT",
}
`;

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 7`] = `
Object {
  "payload": Object {
    "results": Array [],
  },
  "type": "TABLE_INIT",
}
`;

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 8`] = `
Object {
  "payload": Object {
    "results": Array [],
  },
  "type": "TABLE_INIT",
}
`;

exports[`ModelsTableActions updateSearch should dispatch INIT_TABLE 9`] = `
Object {
  "payload": Object {
    "results": Array [
      Object {
        "name": "model-1",
        "vendor_class": "vc1",
      },
    ],
  },
  "type": "TABLE_INIT",
}
`;
